brainiak/utils/fmrisim.py

```diff
@@ -519,8 +666,241 @@ def apply_signal(signal_function,
     return signal
 
 
-def _generate_noise_system(dimensions,
-                           sigma=1.5,
+def _calc_fwhm(volume, mask):
+
+    """ Calculate the FWHM of a volume
+    Takes in a volume and mask and outputs the FWHM of each TR of this
+    volume for the non-masked voxels
+
+    Parameters
+    ----------
+    volume : multidimensional array
+    Functional data to have the FWHM measured. Can be 3d or 4d data
+
+    mask : 4 dimensional array
+    A mask of the voxels to have the FWHM measured from
+
+    Returns
+    -------
+
+    float, list
+    Returns the FWHM of each TR
+
+    """
+
+    # What are the dimensions of the volume
+    dimensions = volume.shape
+
+    # Identify the number of TRs
+    if len(dimensions) == 3:
+        trs = 1
+    else:
+        trs = dimensions[3]
+        dimensions = dimensions[0:3]
+        volume_all = volume  # Store for later
+
+    # Preset size
+    fwhm = np.zeros(trs)
+
+    # Iterate through the TRs, creating a FWHM for each TR
+
+    for tr_counter in list(range(0, trs)):
+
+        # Preset
+        sumSqDer = [0.0, 0.0, 0.0]
```

Comment:
Follow the PEP 8 naming convention:
https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables
`sum_sq_der`? A comment explaining the name would be good.